src/Form/AuthenticationForm.php

```diff
@@ -167,13 +134,27 @@ protected function getEditableConfigNames() {
 
   /**
    * {@inheritdoc}
-   *
-   * @throws \Drupal\Core\Entity\EntityStorageException
-   * @throws \Drupal\key\Exception\KeyValueNotSetException
    */
   public function buildForm(array $form, FormStateInterface $form_state) {
-    $form = parent::buildForm($form, $form_state);
+    // Save the key for later use.
+    $this->activeKey = $this->getActiveKey();
 
+    /** @var \Drupal\apigee_edge\Plugin\KeyProviderRequirementsInterface $key_provider */
+    if (($key_provider = $this->activeKey->getKeyProvider()) instanceof KeyProviderRequirementsInterface) {
+      try {
+        $key_provider->checkRequirements($this->activeKey);
+      }
+      catch (KeyProviderRequirementsException $exception) {
+        $this->messenger()->addError($this->t("The requirements of the selected key provider (@key_provider) are not fulfilled. Fix the errors described below or <a href=':key_config_uri' target='_blank'>change the active key's provider</a>.", [
+          '@key_provider' => $this->activeKey->getKeyProvider()->getPluginDefinition()['label'],
+          ':key_config_uri' => Url::fromRoute('entity.key.edit_form', ['key' => $this->activeKey->id()])->toString(),
+        ]));
+        $this->messenger()->addError($exception->getTranslatableMarkupMessage());
+        return $form;
+      }
+    }
+
+    $form = parent::buildForm($form, $form_state);
```

Comment:
I think parent call could still happen at the beginning of the method. (Unless it adds something unwanted.)

Reply:
I've moved the parent call after the key provider requirement check because that adds the `actions` container (and we should hide that if the key provider requirements are not fulfilled).

Reply:
I think it would be better to hide that with `#access => FALSE` instead of completely removing the array elements that should be there in the `$form` render array.